modules/light-clients/08-wasm/keeper/snapshotter_test.go

```diff
@@ -0,0 +1,100 @@
+package keeper_test
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/keeper"
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/testing/simapp"
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+)
+
+func TestSnapshotter(t *testing.T) {
+	testCases := []struct {
+		name      string
+		wasmFiles []string
+	}{
+		{
+			name:      "single contract",
+			wasmFiles: []string{"../test_data/ics10_grandpa_cw.wasm.gz"},
+		},
+
+		{
+			name:      "multiple contract",
```

Comment:
duplicate contract test case? Or are we assuming a node would never be provided such a snapshot?

Reply:
duplicate contract test would fail with the current test structure on the initial store code call in 08-wasm here: https://github.com/cosmos/ibc-go/blob/feat/wasm-clients/modules/light-clients/08-wasm/keeper/keeper.go#L103-L105

We might be able to work around it tho.

But that being said, all calls to the vm `StoreCode` are idempotent so it should just behave as if its a no-op if its ever reached.